src/deltaCoffeeShop.js

```diff
@@ -0,0 +1,59 @@
+/** Contains the deserialized information of a coffee shop + the distance to it relative to current position */
+class DeltaCoffeeShop {
```

Comment:
I like how you wrote the class, it's orderly, easy to read, documented using Javadocs if I'm not mistaken, and does the job well. However, I don't see why it had to be a class, this seems like it does two things. It represents the `struct` of coffee shop and its distance to the user, so just a data structure, but it also deals with calculating the distance to the user. Do you think distance to something is an intrinsic property of a shop? Or should distance calculation be someone else's responsibility.

Reply:
“Is distance to something an intrinsic property of a shop?” -> no. But it is to a deltaCoffeeShop. I assume you meant to have the coffee shop list calculate the delta when needed, but I don’t really like this approach…
For this, I come with the following observation: the workflow of the app is basically: start -> do something fast -> exit. Thus, it’s safe to assume that X and Y are static for the entire duration of the app. If so, it doesn’t make sense to calculate the delta each time we need it, because that would be redundant in case we need it more than once(i.e. if we later add another coffeeshop). Instead, I implemented the deltaCoffeeShop wrapper, which calculates the delta once and stores it for future use.

Reply:
Good observation.

Reply:
Good observation.